wagtail/admin/views/bulk_action.py

```diff
@@ -100,16 +99,22 @@ def get_context_data(self, **kwargs):
         for obj in objects:
             _objects.append(self.object_context(obj))
         return {
+            **super().get_context_data(**kwargs),
             '{}s'.format(self.object_key): _objects,
             '{}s_with_no_access'.format(self.object_key): objects_with_no_access,
             'next': self.next_url,
-            'submit_url': self.request.path + '?' + self.request.META['QUERY_STRING']
+            'submit_url': self.request.path + '?' + self.request.META['QUERY_STRING'],
         }
 
     def prepare_action(self, objects):
         return
 
-    def post(self, request):
+    def get_execution_context(self):
+        return {}
+
+    def form_valid(self, form):
+        request = self.request
+        self.cleaned_form = form
```

Comment:
nitpick: this is used to pass particular parameters (`include_descendants`, `move_applicable`, `chooser`). I would prefer we either set those directly in each class, or.. let's make this `self.form` cc @jacobtoppm

Reply:
are you suggesting to rename `cleaned_form` to `form`?